packages/web/contexts/GameContext.tsx

```diff
@@ -0,0 +1,328 @@
+/* eslint-disable no-underscore-dangle */
+import { Text, useToast, VStack } from '@metafam/ds';
+import { httpLink } from '@metafam/utils';
+import type {
+  GameProperties,
+  GamePropertiesType,
+  IGameContext,
+  IGameState,
+} from 'components/Landing/OnboardingGame/gameTypes';
+import { MetaLink } from 'components/Link';
+import { CONFIG } from 'config';
+import { Contract } from 'ethers';
+import { useWeb3 } from 'lib/hooks';
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+
+import gameJson from '../components/Landing/OnboardingGame/metagame-onboarding-game.json';
+import ABI from '../contracts/BulkDisbursableNFTs.abi';
+import { get, remove, set } from '../lib/store';
+
+export const GameContext = React.createContext<IGameContext>({
+  game: {
+    name: '',
+    startingElement: '',
+    assets: {},
+    elements: {},
+    jumpers: {},
+    attributes: {},
+    connections: {},
+    components: {},
+  },
+  gameState: () => null,
+  handleChoice: async () => undefined,
+  resetGame: () => false,
+  // typeText: () => '',
+  fetchGameData: async () => {},
+  visitedElements: () => '0',
+  mintChiev: async () => '',
+  connect: async () => undefined,
+  disconnect: async () => undefined,
+  loading: true,
+  txLoading: false,
+  account: '',
+  network: '0x013881',
+  connected: false,
+  connecting: false,
+});
+
+export const GameContextProvider: React.FC = ({ children }) => {
+  const [isLoading, setIsLoading] = useState(true);
+  const [txLoading, setTxLoading] = useState(false);
+  const [gameDataState, setGameDataState] = useState<GamePropertiesType>();
+  const {
+    address,
+    provider,
+    connect,
+    connecting,
+    connected,
+    chainId,
+    disconnect,
+  } = useWeb3();
+  const [account, setAccount] = useState<any>(address ?? '');
+  const toast = useToast();
+
+  /** Function to async fetch `CONFIG.onboardingGameDataURL` as json and return the data
+   * TODO: this needs the func from the main Game.tsx file to be moved here to replace
+   * this function
+   */
+  const fetchGameData = useCallback(async () => {
+    try {
+      setIsLoading(true);
+      console.log('Fetchng GameData...');
+
+      // const response = await fetch(gameJson);
+      // const data = (await response.json()) as GameProperties;
+      const data = gameJson as GameProperties;
+      console.log('fetchGameData', data);
+
+      if (data) {
+        setGameDataState(data);
+      }
+      setIsLoading(false);
+    } catch (error) {
+      console.error(error);
+      setIsLoading(false);
+    }
+  }, []);
+
+  useEffect(() => {
+    if (address) {
+      setAccount(address);
+    }
+  }, [address]);
```

Comment:
Why do we have an extra state variable `account` and this useEffect. Can we not use `address` itself ?

Reply:
i think this has happened because when i was building initially with ThirdWeb (chiev.es roles weren't quite ready), they use `account` and it stuck in places. Agreed, it is not good and I'll get it sorted. 👍🏻